lib/combine_pdf/decrypt.rb

```diff
@@ -76,7 +76,7 @@ def decrypt
 
     def set_general_key(password = '')
       # 1) make sure the initial key is 32 byte long (if no password, uses padding).
-      key = (password.bytes[0..32].to_a + @padding_key)[0..31].to_a.pack('C*').force_encoding(Encoding::ASCII_8BIT)
+      key = (password.bytes[0..32].to_a + @padding_key)[0..31].to_a.pack('C*').dup.force_encoding(Encoding::ASCII_8BIT)
```

Comment:
This dup is redundant and will use up a bunch of memory. The dynamic String returned from `Array#pack` is mutable.

Reply:
Agreed.